src/extensions/huggingface.rs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Huggingface Integration: [HuggingFaceExtension]
+
+use crate::config::ExecutionConfig;
+use crate::extensions::{DftSessionStateBuilder, Extension};
+use log::info;
+use std::sync::Arc;
+
+use opendal::{services::Huggingface, Builder, Operator};
+use url::Url;
+
+#[derive(Debug, Default)]
+pub struct HuggingFaceExtension {}
+
+impl HuggingFaceExtension {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait::async_trait]
+impl Extension for HuggingFaceExtension {
+    async fn register(
+        &self,
+        config: ExecutionConfig,
+        builder: &mut DftSessionStateBuilder,
+    ) -> datafusion_common::Result<()> {
+        let Some(object_store_config) = &config.object_store else {
+            return Ok(());
+        };
+
+        let Some(huggingface_configs) = &object_store_config.huggingface else {
+            return Ok(());
+        };
+
+        for huggingface_config in huggingface_configs {
+            // I'm not that famliar with Huggingface so I'm not sure what permutations of config
+            // values are supposed to work.
+
+            let mut base_url = String::from("https://huggingface.co/");
```

Comment:
I'm guessing we can register url as `hf://datasets/<repo_id>/`. And visit the file in the way `hf://datasets/HuggingFaceTB/finemath/finemath-3plus/train-00000-of-00128.parquet`.

There may be some tricks things on the url handling inside datafusion and oebjct_store.